collect_app/src/main/java/org/odk/collect/android/utilities/PermissionUtils.java

```diff
@@ -88,13 +90,50 @@ public void onPermissionRationaleShouldBeShown(List<PermissionRequest> permissio
                 .check();
     }
 
+    public static void requestCameraPermission(@NonNull Activity activity, @NonNull PermissionListener action) {
+        com.karumi.dexter.listener.single.PermissionListener permissionListener = new com.karumi.dexter.listener.single.PermissionListener() {
+            @Override
+            public void onPermissionGranted(PermissionGrantedResponse response) {
+                action.granted();
+            }
+
+            @Override
+            public void onPermissionDenied(PermissionDeniedResponse response) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(activity, R.style.Theme_AppCompat_Light_Dialog);
+
+                builder.setTitle(R.string.camera_runtime_permission_denied_title)
+                        .setMessage(R.string.camera_runtime_permission_denied_desc)
+                        .setPositiveButton(android.R.string.ok, (dialogInterface, i) -> action.denied())
+                        .setCancelable(false)
+                        .setIcon(R.drawable.ic_photo_camera)
+                        .show();
+            }
+
+            @Override
+            public void onPermissionRationaleShouldBeShown(PermissionRequest permission, PermissionToken token) {
+                token.continuePermissionRequest();
+            }
+        };
+
+        Dexter.withActivity(activity)
+                .withPermission(
+                        Manifest.permission.CAMERA
+                ).withListener(permissionListener)
+                .withErrorListener(error -> Timber.i(error.name()))
+                .check();
+    }
+
     public static boolean checkIfStoragePermissionsGranted(Context context) {
         int read = ContextCompat.checkSelfPermission(context, Manifest.permission.READ_EXTERNAL_STORAGE);
         int write = ContextCompat.checkSelfPermission(context, android.Manifest.permission.WRITE_EXTERNAL_STORAGE);
 
         return read == PackageManager.PERMISSION_GRANTED && write == PackageManager.PERMISSION_GRANTED;
     }
 
+    public static boolean checkIfCameraPermissionGranted(Context context) {
```

Comment:
I am suddenly realizing that `PermissionUtils` provides a bunch of static methods. Should have noticed/brought this up earlier. This means testing methods that require permissions will be impossible, right? It's definitely a good thing to want to reduce redundancy but perhaps there's another structure that would reduce the coupling between activity code and permissions code? CC @jd-alexander

Reply:
Testing methods will be okay because there's a [permission rule](https://github.com/opendatakit/collect/blob/c21d5137608c437261a3a65e05e9fec818daf061/collect_app/src/androidTest/java/org/odk/collect/android/AllWidgetsFormTest.java#L113) that was introduced that grants all the permissions before code is run. 
So the usage of these static methods will always pass. Currently that's the recommended approach by Espresso developers.

Reply:
Perfect, thanks a lot for that clarification.